### PR TITLE
Various fixes to OpenAPI spec

### LINKED ISSUE #181: clicking on "view specific sections for a course" in docs brings you to wrong section
https://user-images.githubusercontent.com/31628143/114979333-ae173800-9e58-11eb-8776-492a21fdffe8.mov

### LINKED ISSUE #183: document getting a course or section for a specific semester
you can get a course or section for a specific semester, but this isn't mentioned anywhere in the docs.

https://api.umd.io/v1/courses/sections/ENGL101-0101?semester=202008

### PR DESCRIPTION
**Pull request by McIntireEvan**
* Fix operationId for /courses/sections/{section_ids} (fixes #181)
* Move course_id and section_id params to generics; add examples (partially #182)
* Add semesterParam in all places that needs it (fixes #183)
* Add command for loading dev docker-compose image

Need a follow-up issue to finish fully documenting all of the course endpoints.

### CODE REVIEW

**Comment by tybug** [comment]
This won't actually build on github pages until we push a new tag, right? Might be nice to build the docs on every pr unless it's really slow (>1min) to do so, since I can see us wanting to fix typos (or add stuff like this) where we don't want to tag a new build

**Comment by McIntireEvan** [comment]
> 
> 
> This won't actually build on github pages until we push a new tag, right? Might be nice to build the docs on every pr unless it's really slow (>1min) to do so, since I can see us wanting to fix typos (or add stuff like this) where we don't want to tag a new build

iirc I set the rule that way to avoid it polluting the commit history. Let me try building it manually and just including that

**Comment by tybug** [comment]
ah right, I guess we'd be committing to master every pr then, I'm used to hosting docs on a `gh-pages` branch where I don't really care about commit history.

I'll look into getting it to build whenever we modify that yaml file

File: README.md [hunk 1]
```diff
@@ -58,21 +60,20 @@ We use Ruby's built-in logger to output messages to standard output. Learn more
...
 [2018-10-18 01:35:02] INFO  (courses_scraper): 178 department/semesters so far
 [2018-10-18 01:35:02] INFO  (courses_scraper): Searching for courses in term 201805
 [2018-10-18 01:35:03] INFO  (courses_scraper): 301 department/semesters so far
 ```
 
-The formatting for outputted messages is as follows:`[DATE TIME] LOG_LEVEL (PROGRAM_NAME): {MESSAGE}`
+The formatting for outputted messages is as follows: `[DATE TIME] LOG_LEVEL (PROGRAM_NAME): {MESSAGE}`
 
-An example of a log call in ruby:
-`logger.info(prog_name) {"MESSAGE"}`
```
Comment (DonIsaac) [inline README.md:71]: if you're gonna explicityly say this is a ruby example and put it on a new line, why not do something like
```ruby
logger.info(prog_name) {"MESSAGE"}
```
?

Comment (DonIsaac) [reply README.md:71]: w8 this is old code, but I think (?) it could still apply?

File: README.md [hunk 2]
```diff
@@ -58,21 +60,20 @@ We use Ruby's built-in logger to output messages to standard output. Learn more
...
+An example of a log call in ruby: `logger.info(prog_name) {"MESSAGE"}`
 
 You should use Ruby's built-in log-levels where appropriate, when displaying errors you should use logger.error, when displaying information you should use logger.info, and so on.
 
-Our logger implementation is located at the `scraper_common.rb` file located at `$app/scraper_common.rb`
+Our logger implementation is located at the `scraper_common.rb` file located at `$app/scraper_common.rb`.
```
Comment (DonIsaac) [inline README.md:76]: it's actually `app/scrapers/scraper_common.rb`

File: README.md (hunk 1 above)
Comment (McIntireEvan) [reply README.md]: Didn't actually touch this

File: README.md (hunk 2 above)
Comment (McIntireEvan) [reply README.md]: Didn't actually touch this, but can fix

**Review by tybug — APPROVED** [review]

**Pull request merged**